consensus/src/genesis/genesis_init.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use mina_rs_base::types::*;
+
+/// Trait for genesis block initialization logic
+/// # Example
+/// ```
+/// use mina_rs_base::types::*;
+/// use mina_consensus::genesis::{GenesisInit, DEVNET_CONFIG, MAINNET_CONFIG};
+/// let genesis_mainnet = ExternalTransition::init_genesis(&MAINNET_CONFIG);
+/// let genesis_devnet = ExternalTransition::init_genesis(&DEVNET_CONFIG);
+/// ```
+pub trait GenesisInit {
+    fn init_genesis(config: &GenesisInitConfig) -> ExternalTransition;
+}
+
+impl GenesisInit for ExternalTransition {
```

Comment:
minor, but i think we can call this trait simply `Genesis` and have a method name `from_genesis` instead of `init_genesis`, as `from_*` is the usual method naming convention in Rust codebase, when deriving a value out of some other value. Thoughts?

Reply:
How about from_genesis_config ?

Reply:
yeah

Reply:
fixed